File: zigzagConversion.js
var convert = function(s, numRows) {
    let result = [];
    for(let i = 0; i<numRows; i++){
        result.push("")
    }
    let index = 0;
    let step = 1;
    for(let i = 0; i<s.length; i++){
        let char = s[i];
        result[index] += char;
        if(index === 0){
            step = 1;
        } else if(index === numRows-1){
            step = -1;
        }
        index += step;
    }
    return result.join("");
};
const s = "PAYPALISHIRING";
const numRows = 3;
console.log("convert", convert(s, numRows));
//paypalishiring
// 
// p            i           n
// a        l   s       i   g
// y    a       h   r
// p            i

File: zigzagConversion.cs
public class Solution
{
    public string Convert(string s, int numRows)
    {
        if (numRows == 1 || numRows >= s.Length) return s;
        var gap = 2 * numRows - 2;
        var length = s.Length;
        var sb = new StringBuilder(s.Length);

        for (int i = 0; i < numRows; i++)
        {
            for (int j = 0; j + i < s.Length; j += gap)
            {
                if (j + i < s.Length)
                {
                    sb.Append(s[j + i]);
                    if (i != 0 && i != numRows - 1 && (j + gap - i < s.Length))
                    {
                        sb.Append(s[j + gap - i]);
                    }
                }

            }

        }
        return sb.ToString();
    }
}